From b3077532edcfcd40c4acc89f436c227e2656fec3 Mon Sep 17 00:00:00 2001
From: Jacob Vaverka <jacob.vaverka@juliahub.com>
Date: Mon, 21 Nov 2022 14:58:31 -0600
Subject: [PATCH 32/41] style: use atom-one-dark theme

---
 _quarto.yml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/_quarto.yml b/_quarto.yml
index f9b9923..9db5fef 100644
--- a/_quarto.yml
+++ b/_quarto.yml
@@ -55,6 +55,8 @@ format:
   pdf:
     documentclass: scrreprt
 
+highlight-style: atom-one
+
 jupyter: scimlbook-1.8
 
 execute:
-- 
2.33.1.windows.1

