From 95ebe00b46e1634ee44c742044f8445a4800b17b Mon Sep 17 00:00:00 2001
From: Jacob Vaverka <jacob.vaverka@juliahub.com>
Date: Tue, 22 Nov 2022 07:45:38 -0600
Subject: [PATCH 41/41] feat: ignore jupyter checkpoints

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 2fd233c..a86b87c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -11,4 +11,5 @@ code/
 .quarto/
 JuliaSysimage.so
 JuliaSysimage.dll
+.ipynb_checkpoints/
 /.quarto/
-- 
2.33.1.windows.1

